### PR TITLE
brotli-compress .socket.facts.json on upload

### PR DESCRIPTION
**Pull request by barslev**
## Summary

Brotli-compress `.socket.facts.json` at upload time. Coana keeps writing plain JSON; only the wire bytes between socket-cli and depscan change. ~85% reduction on the simple-npm fixture; production mono-repo payloads see 60+ MB → ~10 MB.

## How

- New `compressSocketFactsForUpload(scanPaths)` in `src/utils/coana.mts` — async + streaming (`createReadStream → createBrotliCompress → createWriteStream` via `pipeline`). For any `.socket.facts.json` in scanPaths, writes a sibling `.socket.facts.json.br` next to the source file. Returns swapped paths + cleanup callback.
- `handle-create-new-scan.mts` wraps `fetchCreateOrgFullScan` in `try { compress; upload; } finally { cleanup; }`.

## Why sibling-write (not `os.tmpdir()`, not a temp subdir)

The SDK passes `path.relative(cwd, brPath)` as the multipart name. depscan's `addStreamEntry` rejects entries with `..` traversal — a tmpdir path resolves to `../../../var/folders/...` and gets dropped to `unmatchedFiles`. Sibling-write keeps the relative path inside cwd, and keeps directory shape symmetric with the plain `.socket.facts.json` upload (depscan strips the `.br` suffix at ingest, so `<dir>/.socket.facts.json.br` and `<dir>/.socket.facts.json` resolve to the same storage path).

Concurrent scans against the same source dir already collide on `.socket.facts.json` itself (coana writes to a single path), so the sibling `.br` doesn't introduce a new race.

## Tests

- 16 unit cases in `src/utils/coana.test.mts` (sibling-path round-trip, passthrough, idempotent cleanup that leaves the source intact, plus the existing `extractTier1ReachabilityScanId` / `extractReachabilityErrors` matrix).
- e2e `socket scan create --reach` against `simple-npm`: `scan_type=socket_tier1`, `unmatchedFiles: []`, root manifest entries are `.socket.facts.json` and `package.json` (storage path matches the plain upload — dir-preservation symmetry holds).

---

> [!NOTE]
> **Medium Risk**
> Changes the scan creation upload pipeline to rewrite `.socket.facts.json` inputs into temporary Brotli-compressed `.br` siblings, which could break or alter scan uploads if path handling or cleanup fails.
> 
> **Overview**
> **Scan uploads now Brotli-compress Coana facts files.** `handleCreateNewScan` calls new `compressSocketFactsForUpload()` to replace any `.socket.facts.json` entries in `scanPaths` with sibling `.socket.facts.json.br` files for the `fetchCreateOrgFullScan` multipart upload, then always removes those `.br` files in a `finally` block.
> 
> **Adds a streaming compressor utility + tests.** `utils/coana.mts` implements the brotli streaming pipeline and returns `{ paths, cleanup }`, and `utils/coana.test.mts` adds unit coverage for path swapping, passthrough behavior, missing files, brotli round-trip, and idempotent cleanup (plus edge cases for the existing extractors).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit bb966857067305d13c4252d51856194bafab4ce5. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by jdalton** [comment]
This is rad!

**Review by mtorp — APPROVED** [review]
LGTM ✅

File: src/utils/coana.mts [hunk 1]
```diff
@@ -11,8 +16,84 @@
...
+      const brPath = `${p}.br`
+      await pipeline(
+        createReadStream(p),
+        createBrotliCompress(),
+        createWriteStream(brPath),
+      )
+      brPaths.push(brPath)
+      return brPath
+    }),
+  )
```
[the remark this answers is not in the document]
Comment (barslev) [reply src/utils/coana.mts:104]: Addressed in dbed10b44.

Three changes that together make the helper failure-safe:

- Track each `.br` path in `brPaths` **before** `pipeline()` starts, so partials are visible to cleanup. (`rm({ force: true })` no-ops on missing paths.)
- Switch the parallel batch from `Promise.all` to `Promise.allSettled` — with `all`, the first rejection bubbled out while sibling pipelines were still writing, and cleanup would race the live writes. `allSettled` lets every pipeline finish (success or fail) before we look for a failure.
- On failure, run `cleanup()` internally before re-throwing, so the caller (which only gets a chance to call cleanup on success) doesn't have to handle this case.

Added a regression test (`removes partial .br siblings if compression fails mid-batch`) that pre-occupies one entry's `.br` destination with a directory to force the pipeline to reject mid-batch, then asserts the other entry's `.br` is not orphaned.
